jwql/instrument_monitors/common_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,727 @@
+                                           }
+                    self.stats_table.__table__.insert().execute(cosmic_ray_db_entry)
+
+                    logging.info("Successfully inserted into database. \n")
+                except:
+                    logging.info("Could not insert entry into database. \n")
+
+                # Delete fits files in order to save disk space
+                logging.info("Removing pipeline products in order to save disk space. \n")
+                try:
+                    shutil.rmtree(self.obs_dir)
+                except OSError as e:
+                    logging.error(f"Unable to delete {self.obs_dir}")
+                    logging.error(e)
+
+                # Remove initial copy of input file as well
+                os.remove(file_name)
+
+    def pull_filenames(self, file_info):
+        """Extract filenames from the list of file information returned from
+        query_mast.
+
+        Parameters
+        ----------
+        file_info : dict
+            Dictionary of file information returned by ``query_mast``
+
+        Returns
+        -------
+        files : list
+            List of filenames (without paths) extracted from ``file_info``
+        """
+        files = [element['filename'] for element in file_info['data']]
+        return files
+
+    @log_fail
+    @log_info
+    def run(self):
+        """The main method. See module docstrings for additional info
+
+        Queries MAST for new MIRI data and copies it to a working
+        directory where it is run through the JWST pipeline. The output
+        of the 'jump' and 'rate' steps is used to determine the number
+        and magnitudes of cosmic rays which is then saved to the
+        database.
+        """
+
+        logging.info('Begin logging for cosmic_ray_monitor')
+
+        self.query_end = Time.now().mjd
+
+        for instrument in ['nircam']:  #JWST_INSTRUMENT_NAMES:
+            if instrument == 'miri' or instrument == 'nircam':
+                self.instrument = instrument
+
+                # Identify which tables to use
+                self.identify_tables()
+
+                # Get a list of possible apertures
+                #possible_apertures = list(Siaf(instrument).apernames)
+                possible_apertures = self.possible_apers(instrument)
+
+                #XXXFOR TESTING
+                possible_apertures = ['NRCA1_FULL']
+
+                # Use this line instead to save time while testing
+                #possible_apertures = ['MIRIM_FULL', 'NRCB4_FULL']
+
+                for aperture in possible_apertures:
+
+                    logging.info('')
+                    logging.info('Working on aperture {} in {}'.format(aperture, instrument))
+
+                    self.aperture = aperture
+
+                    # We start by querying MAST for new data
+                    self.query_start = self.most_recent_search()
+
+                    #XXXXFOT TESTING
```

Comment:
Use the most recent query date as start date here.

Reply:
Done